apps/api/src/app/partner-integrations/usecases/update-vercel-configuration/update-vercel-configuration.usecase.ts

```diff
@@ -0,0 +1,261 @@
+      const newData = {
+        privateKey: curr.apiKeys[0].key,
+        clientKey: curr.identifier,
+        updateProjectDetails: updateProjectDetails.filter((detail) => projectIds.includes(detail.projectId)),
+        addProjectIds: projectIds.filter((id) => addProjectIds.includes(id)),
+      };
+      acc[curr._organizationId] = newData;
+
+      return acc;
+    }, {});
+
+    return mappedData;
+  }
+
+  private async setEnvironments({ clientKey, projectIds, privateKey, teamId, token }: ISetEnvironment): Promise<void> {
+    const projectApiUrl = `${process.env.VERCEL_BASE_URL}/v9/projects`;
+    const target = ['production', 'preview', 'development'];
+    const type = 'encrypted';
+
+    const apiKeys = [
+      {
+        target,
+        type,
+        value: clientKey,
+        key: 'NOVU_CLIENT_APP_ID',
+      },
+      {
+        target,
+        type,
+        value: privateKey,
+        key: 'NOVU_API_SECRET',
+      },
+    ];
+
+    await Promise.all(
+      projectIds.map((projectId) =>
+        lastValueFrom(
+          this.httpService.post(`${projectApiUrl}/${projectId}/env${teamId ? `?teamId=${teamId}` : ''}`, apiKeys, {
+            headers: {
+              Authorization: `Bearer ${token}`,
+              'Content-Type': 'application/json',
+            },
+          })
+        )
+      )
+    );
+  }
+
+  private async getVercelProjects(accessToken: string, teamId: string | null, projectIds: string[]) {
+    const response = await lastValueFrom(
+      this.httpService.get(`${process.env.VERCEL_BASE_URL}/v4/projects${teamId ? `?teamId=${teamId}` : ''}`, {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      })
+    );
+
+    return this.mapProjects(response.data.projects, projectIds);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private mapProjects(projects: any[], projectIds: string[]) {
+    return projects.reduce<MappedProjectData>(
+      (acc, curr) => {
+        const id = curr.id;
+        if (projectIds.includes(id)) {
+          const vercelEnvs = curr?.env;
+          const clientEnv = vercelEnvs?.filter((e) => e.key === 'NOVU_CLIENT_APP_ID');
+          const secretEnv = vercelEnvs?.filter((e) => e.key === 'NOVU_API_SECRET');
+
+          if (clientEnv.length > 0 && secretEnv.length > 0) {
+            acc.updateProjectDetails.push({
+              projectId: id,
+              clientEnvId: clientEnv[0].id,
+              secretEnvId: secretEnv[0].id,
+            });
+          } else {
+            acc.addProjectIds.push(id);
+          }
+        }
+
+        return acc;
+      },
+      { updateProjectDetails: [], addProjectIds: [] }
+    );
+  }
+
+  private async saveProjectIds(command) {
+    await this.organizationRepository.bulkUpdatePartnerConfiguration(
+      command.userId,
+      command.data,
+      command.configurationId
+    );
+  }
+
+  private async updateEnvironments({
+    clientKey,
```

Comment:
Need to make individual call for each `env` to be updated for the `updateProjectDetails`,